### ISSUE TITLE
enhancement - automated workflow (status change from commits)

### ISSUE THREAD

**Issue by tiny-dancer** [taigaio/taiga-back#217]
It would be powerful if it was possible to build on the "CHANGING ELEMENTS STATUS VIA COMMIT MESSAGE" feature to be able to dictate within taiga what happens when a issue reference within a commit message is present.

A couple examples of setting the automated workflow within taiga:

simple example:
When TG-1 commit is present in repo --> TG-1 status change to "in progress"

a slightly less example in a gitflow environment:
When TG-1 commit is present in feature/ branch --> TG-1 status change to "dev in progress"
When TG-1  commit is present in dev/ branch --> TG-1 status change to "dev complete" 
When TG-1  commit is present in master/ branch --> TG-1 status change to "closed" (a fail safe as i'm sure this would be tested and closed by now) 

Another building block on this would be if the TG-1 commit was found within a pull/merge request:
When TG-1  commit is present in pull request --> TG-1 status change to "ready for review" 

I skimmed through the issues prior to posting this, apologies if this has already been documented.

**Comment by bameda** [comment]
Hi @tiny-dancer, first of all thanks for using Taiga and for your feedback.
Do you know about github/gitlab/bitbucket integration in Taiga? https://taiga.io/support/changing-elements-status-via-commit-message/

**Comment by tiny-dancer** [comment]
Thanks @bameda, it's a great product and hopefully someday i can contribute if i have time (and learn python lol)

I am aware of that feature and because of that feature the enhancement above could probably be de-prioritized.  My suggestion referred to extending that capability to where you can set the status changes from within the taiga frontend (not the commit messages).  

For example, in taiga you set that the first time an issue reference (TG-*) is pushed/present in the repo - TG-* 's status will change to in progress. Therefore everytime TG-* is pushed to the repo that status will move to in progress (if in a lower status such as new).  This example would avoid having to "#in-progress" every time pushing a new feature.

You would no longer need to reference the status slug in the commit message and the logic the admin has set in the taiga app will handle the status changes.

**Comment by bameda** [comment]
Ok, I know. It's a good enhancement I'm going to add for be studied for the Taiga team. But personally, like one rule of the zen of python, I prefer to be explicit than implicit and set the new status in the commit message.

Thank you very much.

**Comment by superalex** [comment]
I prefer the explicit approach too :)

One of the "problems" we have now is that if we have a "commit with action" that action is executed in all the branches that commit appear (executing multiple times the action if you merge branches)

If we add support for multiple actions depending on the branch we could have a message like:

TG-X feature-branch#dev-in-progress dev#dev-complete master#closed

**Comment by tiny-dancer** [comment]
Yup that would work, but could get a little repetitive. It's like code duplication why type #in-progress every commit when you can do it once for all.  I feel the explicit approach works great for kicking off task development, but it's not often in a status workflow that a commit would trigger statuses such as being closed, especially before hitting a ci server.  But as i'm typing this out, in most workflows the explicit approach will work great, it's just a "closed" status that would be missing. I guess i'm just dreaming for the day where task statuses will be automated by where they are in the development process (git through deployments), cheers to being lazy :smile:

**Comment by superalex** [comment]
Hahahahahahaha, laziness makes the world improve!. We have to review this deeply and decide what would be the better way to make this or at least what would be the second version for it that improves what we have. Thank you very much for your suggestions :+1:

**Comment by tiny-dancer** [comment]
haha no problem, thanks for the consideration

**Comment by cmoralesweb** [comment]
Hey @superalex , regarding the problem you mentioned in this comment https://github.com/taigaio/taiga-back/issues/217#issuecomment-70456278

We've suffered that error, having issues changing their status inexpectedly after a merge (a closed issue gets marked as in-progress , for example). 

I haven't found that as an issue here, should I report it, do you not consider it an error...?

Thanks!
